components/canvas/webgl_thread.rs

```diff
@@ -115,6 +120,27 @@ impl<VR: WebVRRenderHandler + 'static> WebGLThread<VR> {
     /// Handles a generic WebGLMsg message
     #[inline]
     fn handle_msg(&mut self, msg: WebGLMsg, webgl_chan: &WebGLChan) -> bool {
+        let target_id = match msg {
+            WebGLMsg::CreateContext(..) |
+            WebGLMsg::Lock(..) |
+            WebGLMsg::Unlock(..) |
```

Comment:
We should probably assert that we don't receive lock messages while it's unlocked, or similar. Otherwise the code below in `handle_unlock` won't work. Though I think this is a pre-existing issue.